packages/babel-plugin-remove-undefined-if-possible/src/index.js

```diff
@@ -0,0 +1,82 @@
+"use strict";
+
+function removeRvalIfUndefined(declaratorPath) {
+  const rval = declaratorPath.get("init")
+                              .evaluate();
+  if (rval.confident === true && rval.value === undefined) {
+    declaratorPath.node.init = null;
+  }
+}
+
+function areAllBindingsNotSeen(declaratorPath, seenNames, t) {
+  const id = declaratorPath.node.id;
+  const names = t.getBindingIdentifiers(id);
+  for (const name in names) {
+    if (seenNames.has(name)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+module.exports = function({ types: t }) {
+  let names = null;
+  let functionNesting = 0;
+  return {
+    name: "remove-undefined-if-possible",
+    visitor: {
+      Program: {
+        enter() {
+          names = new Set();
+          functionNesting = 0;
+        },
+        exit() {
+          names = null;
+          functionNesting = 0;
+        },
+      },
+      Function: {
+        enter() {
+          functionNesting++;
+        },
+        exit() {
+          functionNesting--;
```

Comment:
Good point. There's also a concern for a case like:

```javascript
function foo() {
  bar();
  var x = undefined;
  console.log(x);
  function bar() {
    x = 3;
  }
}
```

In which case we don't want to remove `= undefined` (and the cumulative-identifier approach wouldn't work). I'm not currently sure what the best approach to solve this case is.